Mathlib/Tactic/Linarith/Ineq.lean

```diff
@@ -0,0 +1,100 @@
+/-
```

Comment:
There's no longer anything `linarith`-specific in this file and you plan to use it in `linear_inequality`. Should it move somewhere more general, perhaps to `Mathlib/Lean/Expr`?

Reply:
Sure, I can do that.  Should I also remove `Ineq` from the `Linarith` namespace?

Reply:
For a location: How about `Mathlib/Data/Ineq`?  The mathlib material about the similar `Ordering` type (an enum with constructors `| lt | eq | gt`) lives in `Mathlib/Data/Ordering`.

Reply:
I moved to `Mathlib/Data/Ineq` and namespaced the `Ineq` data type to `Mathlib.Ineq`.  I'll merge tomorrow unless I hear from you with alternative suggestions!